### ISSUE TITLE
Upgrade 2022.08 to 2023.01 not working

### ISSUE THREAD

**Issue by umrubi1** [uyuni-project/uyuni#6588]
### Problem description

Hi everyone,

today I have tried the update from 2022.08 to 2023.01 (twice) without luck. After the first startup of spacewalk-service I see a lot of error messages. It seems to be, that the database was not modified correctly. 

Login to  WebUI was not possible anymore after deleting cookies. 

I have seen, that Julio has written in the mailing list, that "_2022.12 will require a special procedure if you are not
upgrading from 2022.11_" but I can't see where my problem is coming from.

Thanks in advance and best regards

### Steps to reproduce

1. VM Snaphot (offline)
2. zypper ref & zypper in zypper
3. spacewalk-service stop
4. zypper up --allow-vendor-change
5. spacewalk-service start // journalctl -xef -u uyuni-check-database.service

### Uyuni version

```bash
# zypper info Uyuni-server-release (from 2022.08)

Information for package Uyuni-Server-release:
---------------------------------------------
Repository     : Uyuni Server Stable
Name           : Uyuni-Server-release
Version        : 2023.01-220400.196.2.uyuni2
Arch           : x86_64
Vendor         : obs://build.opensuse.org/systemsmanagement:Uyuni
Support Level  : Level 3
Installed Size : 1.4 KiB
Installed      : Yes
Status         : out-of-date (version 2022.08-185.2.uyuni2 installed)
Source package : Uyuni-Server-release-2023.01-220400.196.2.uyuni2.src
Summary        : Uyuni Server
Description    :
    Uyuni lets you efficiently manage physical, virtual,
    and cloud-based Linux systems. It provides automated and cost-effective
    configuration and software management, asset management, and system
    provisioning.

# rpm -qa|grepi Uyuni-server-release
Uyuni-Server-release-2022.08-185.2.uyuni2.x86_64
```

### Useful logs

```bash
### First startup of spacewalk-service
# spacewalk-service start
Starting spacewalk services...
  Checking DB schema and running DB schema upgrade if needed. This may take a while.
  Call the following command to see progress: journalctl -f -u uyuni-check-database.service
Done.

---

#### journalctl -xef -u uyuni-check-database.service
Feb 07 16:31:24 uyuni01 systemd[1]: Starting Uyuni check database...
¦¦ Subject: A start job for unit uyuni-check-database.service has begun execution
¦¦ Defined-By: systemd
¦¦ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
¦¦
¦¦ A start job for unit uyuni-check-database.service has begun execution.
¦¦
¦¦ The job identifier is 16428.
Feb 07 16:31:24 uyuni01 spacewalk-startup-helper[58207]: Prototype mismatch: sub Spacewalk::Setup::idn_to_ascii: none vs ($;$$) at /usr/lib/perl5/5.26.1/Exporter.pm line 66.
Feb 07 16:31:24 uyuni01 spacewalk-startup-helper[58207]:  at /usr/lib/perl5/vendor_perl/5.26.1/Spacewalk/Setup.pm line 29.
Feb 07 16:31:24 uyuni01 spacewalk-startup-helper[58215]: report_db_host = uyuni01.vi.vector.int
Feb 07 16:31:24 uyuni01 spacewalk-startup-helper[58219]: Prototype mismatch: sub Spacewalk::Setup::idn_to_ascii: none vs ($;$$) at /usr/lib/perl5/5.26.1/Exporter.pm line 66.
Feb 07 16:31:24 uyuni01 spacewalk-startup-helper[58219]:  at /usr/lib/perl5/vendor_perl/5.26.1/Spacewalk/Setup.pm line 29.
Feb 07 16:31:24 uyuni01 spacewalk-startup-helper[58227]: Prototype mismatch: sub Spacewalk::Setup::idn_to_ascii: none vs ($;$$) at /usr/lib/perl5/5.26.1/Exporter.pm line 66.
Feb 07 16:31:24 uyuni01 spacewalk-startup-helper[58227]:  at /usr/lib/perl5/vendor_perl/5.26.1/Spacewalk/Setup.pm line 29.
Feb 07 16:31:24 uyuni01 spacewalk-startup-helper[58229]: Prototype mismatch: sub Spacewalk::Setup::idn_to_ascii: none vs ($;$$) at /usr/lib/perl5/5.26.1/Exporter.pm line 66.
Feb 07 16:31:24 uyuni01 spacewalk-startup-helper[58229]:  at /usr/lib/perl5/vendor_perl/5.26.1/Spacewalk/Setup.pm line 29.
Feb 07 16:31:24 uyuni01 spacewalk-startup-helper[58234]: Prototype mismatch: sub Spacewalk::Setup::idn_to_ascii: none vs ($;$$) at /usr/lib/perl5/5.26.1/Exporter.pm line 66.
Feb 07 16:31:24 uyuni01 spacewalk-startup-helper[58234]:  at /usr/lib/perl5/vendor_perl/5.26.1/Spacewalk/Setup.pm line 29.
Feb 07 16:31:24 uyuni01 spacewalk-startup-helper[58239]: Prototype mismatch: sub Spacewalk::Setup::idn_to_ascii: none vs ($;$$) at /usr/lib/perl5/5.26.1/Exporter.pm line 66.
Feb 07 16:31:24 uyuni01 spacewalk-startup-helper[58239]:  at /usr/lib/perl5/vendor_perl/5.26.1/Spacewalk/Setup.pm line 29.
Feb 07 16:31:25 uyuni01 spacewalk-startup-helper[58247]: Prototype mismatch: sub Spacewalk::Setup::idn_to_ascii: none vs ($;$$) at /usr/lib/perl5/5.26.1/Exporter.pm line 66.
Feb 07 16:31:25 uyuni01 spacewalk-startup-helper[58247]:  at /usr/lib/perl5/vendor_perl/5.26.1/Spacewalk/Setup.pm line 29.
Feb 07 16:31:55 uyuni01 spacewalk-startup-helper[58227]: Schema upgrade: [susemanager-schema-4.3.13-1.2.uyuni2] -> [susemanager-schema-4.4.4-220400.1.2.uyuni2]
Feb 07 16:31:55 uyuni01 spacewalk-startup-helper[58227]: Searching for upgrade path to: [susemanager-schema-4.4.4-220400]
Feb 07 16:31:55 uyuni01 spacewalk-startup-helper[58227]: Searching for upgrade path to: [susemanager-schema-4.4.4]
Feb 07 16:31:55 uyuni01 spacewalk-startup-helper[58227]: Searching for start path:  [susemanager-schema-4.3.13-1]
Feb 07 16:31:55 uyuni01 spacewalk-startup-helper[58227]: Searching for start path:  [susemanager-schema-4.3.13]
Feb 07 16:31:55 uyuni01 spacewalk-startup-helper[58227]: The path: [susemanager-schema-4.3.13] -> [susemanager-schema-4.4.0] -> [susemanager-schema-4.4.1] -> [susemanager-schema-4.4.2] -> [susemanager-schema-4.4.3] -> [susemanager-schema-4.4.4]
Feb 07 16:31:55 uyuni01 spacewalk-startup-helper[58227]: Planning to run schema upgrade with dir '/var/log/spacewalk/schema-upgrade/schema-from-20230207-163155'
Feb 07 16:31:55 uyuni01 spacewalk-startup-helper[58227]: Executing spacewalk-sql, the log is in [/var/log/spacewalk/schema-upgrade/schema-from-20230207-163155-to-susemanager-schema-4.4.4.log].
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58227]: [2.9K blob data]
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58500]: Prototype mismatch: sub Spacewalk::Setup::idn_to_ascii: none vs ($;$$) at /usr/lib/perl5/5.26.1/Exporter.pm line 66.
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58500]:  at /usr/lib/perl5/vendor_perl/5.26.1/Spacewalk/Setup.pm line 29.
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58227]: The database schema was upgraded to version [susemanager-schema-4.4.4-220400.1.2.uyuni2].
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58505]: Prototype mismatch: sub Spacewalk::Setup::idn_to_ascii: none vs ($;$$) at /usr/lib/perl5/5.26.1/Exporter.pm line 66.
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58505]:  at /usr/lib/perl5/vendor_perl/5.26.1/Spacewalk/Setup.pm line 29.
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58507]: Prototype mismatch: sub Spacewalk::Setup::idn_to_ascii: none vs ($;$$) at /usr/lib/perl5/5.26.1/Exporter.pm line 66.
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58507]:  at /usr/lib/perl5/vendor_perl/5.26.1/Spacewalk/Setup.pm line 29.
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58512]: Prototype mismatch: sub Spacewalk::Setup::idn_to_ascii: none vs ($;$$) at /usr/lib/perl5/5.26.1/Exporter.pm line 66.
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58512]:  at /usr/lib/perl5/vendor_perl/5.26.1/Spacewalk/Setup.pm line 29.
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58517]: Prototype mismatch: sub Spacewalk::Setup::idn_to_ascii: none vs ($;$$) at /usr/lib/perl5/5.26.1/Exporter.pm line 66.
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58517]:  at /usr/lib/perl5/vendor_perl/5.26.1/Spacewalk/Setup.pm line 29.
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58525]: Prototype mismatch: sub Spacewalk::Setup::idn_to_ascii: none vs ($;$$) at /usr/lib/perl5/5.26.1/Exporter.pm line 66.
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58525]:  at /usr/lib/perl5/vendor_perl/5.26.1/Spacewalk/Setup.pm line 29.
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58505]: Schema upgrade: [uyuni-reportdb-schema-4.3.5-1.21.uyuni2] -> [uyuni-reportdb-schema-4.4.1-220400.1.41.uyuni2]
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58505]: Searching for upgrade path to: [uyuni-reportdb-schema-4.4.1-220400]
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58505]: Searching for upgrade path to: [uyuni-reportdb-schema-4.4.1]
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58505]: Searching for start path:  [uyuni-reportdb-schema-4.3.5-1]
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58505]: Searching for start path:  [uyuni-reportdb-schema-4.3.5]
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58505]: The path: [uyuni-reportdb-schema-4.3.5] -> [uyuni-reportdb-schema-4.4.0] -> [uyuni-reportdb-schema-4.4.1]
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58505]: Planning to run schema upgrade with dir '/var/log/spacewalk/reportdb-schema-upgrade/schema-from-20230207-163159'
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58505]: Executing spacewalk-sql, the log is in [/var/log/spacewalk/reportdb-schema-upgrade/schema-from-20230207-163159-to-uyuni-reportdb-schema-4.4.1.log].
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58505]: [157B blob data]
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58542]: Prototype mismatch: sub Spacewalk::Setup::idn_to_ascii: none vs ($;$$) at /usr/lib/perl5/5.26.1/Exporter.pm line 66.
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58542]:  at /usr/lib/perl5/vendor_perl/5.26.1/Spacewalk/Setup.pm line 29.
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58505]: The database schema was upgraded to version [uyuni-reportdb-schema-4.4.1-220400.1.41.uyuni2].
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58552]: Prototype mismatch: sub Spacewalk::Setup::idn_to_ascii: none vs ($;$$) at /usr/lib/perl5/5.26.1/Exporter.pm line 66.
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58552]:  at /usr/lib/perl5/vendor_perl/5.26.1/Spacewalk/Setup.pm line 29.
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58565]: Prototype mismatch: sub Spacewalk::Setup::idn_to_ascii: none vs ($;$$) at /usr/lib/perl5/5.26.1/Exporter.pm line 66.
Feb 07 16:31:59 uyuni01 spacewalk-startup-helper[58565]:  at /usr/lib/perl5/vendor_perl/5.26.1/Spacewalk/Setup.pm line 29.
Feb 07 16:32:00 uyuni01 spacewalk-startup-helper[58578]: Prototype mismatch: sub Spacewalk::Setup::idn_to_ascii: none vs ($;$$) at /usr/lib/perl5/5.26.1/Exporter.pm line 66.
Feb 07 16:32:00 uyuni01 spacewalk-startup-helper[58578]:  at /usr/lib/perl5/vendor_perl/5.26.1/Spacewalk/Setup.pm line 29.
Feb 07 16:32:00 uyuni01 systemd[1]: Finished Uyuni check database.
¦¦ Subject: A start job for unit uyuni-check-database.service has finished successfully
¦¦ Defined-By: systemd
¦¦ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
¦¦
¦¦ A start job for unit uyuni-check-database.service has finished successfully.
¦¦
¦¦ The job identifier is 16428.
----


####### tail -f postgresql-2023-02-07_162348.log
2023-02-07 16:23:48.029 CET   [23705]LOG:  starting PostgreSQL 14.6 on x86_64-suse-linux-gnu, compiled by gcc (SUSE Linux) 7.5.0, 64-bit
2023-02-07 16:23:48.029 CET   [23705]LOG:  listening on IPv4 address "0.0.0.0", port 5432
2023-02-07 16:23:48.030 CET   [23705]LOG:  listening on IPv6 address "::", port 5432
2023-02-07 16:23:48.032 CET   [23705]LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2023-02-07 16:23:48.066 CET   [23705]LOG:  listening on Unix socket "/tmp/.s.PGSQL.5432"
2023-02-07 16:23:48.070 CET   [23707]LOG:  database system was shut down at 2023-02-07 16:23:47 CET
2023-02-07 16:23:48.075 CET   [23705]LOG:  database system is ready to accept connections
2023-02-07 16:31:59.070 CET uyuni uyuni [58499]WARNING:  there is no transaction in progress
2023-02-07 16:32:35.424 CET uyuni uyuni [61856]ERROR:  relation "hibernate_sequence" does not exist at character 17
2023-02-07 16:32:35.424 CET uyuni uyuni [61856]STATEMENT:  select nextval ('hibernate_sequence')
2023-02-07 16:32:35.426 CET uyuni uyuni [61856]ERROR:  current transaction is aborted, commands ignored until end of transaction block
2023-02-07 16:32:35.426 CET uyuni uyuni [61856]STATEMENT:  select 'c3p0 ping' from dual
2023-02-07 16:32:35.436 CET uyuni uyuni [61856]ERROR:  current transaction is aborted, commands ignored until end of transaction block
2023-02-07 16:32:35.436 CET uyuni uyuni [61856]STATEMENT:  select nextval ('hibernate_sequence')
---


#### catalina.log ####
com.redhat.rhn.common.db.WrappedSQLException: ERROR: column "proxy_id" does not exist
  Position: 310
	at com.redhat.rhn.common.translation.SqlExceptionTranslator.postgreSqlException(SqlExceptionTranslator.java:49) ~[rhn.jar:?]
	at com.redhat.rhn.common.translation.SqlExceptionTranslator.sqlException(SqlExceptionTranslator.java:39) ~[rhn.jar:?]
	at com.redhat.rhn.common.db.NamedPreparedStatement.execute(NamedPreparedStatement.java:147) ~[rhn.jar:?]
	at com.redhat.rhn.common.db.datasource.CachedStatement.execute(CachedStatement.java:499) ~[rhn.jar:?]
	at com.redhat.rhn.common.db.datasource.CachedStatement.lambda$executeChecking$1(CachedStatement.java:456) ~[rhn.jar:?]
	at org.hibernate.jdbc.WorkExecutor.executeReturningWork(WorkExecutor.java:55) [hibernate-core-5.jar:5.3.25.Final]
	at org.hibernate.internal.SessionImpl$2.accept(SessionImpl.java:2421) [hibernate-core-5.jar:5.3.25.Final]
	at org.hibernate.engine.jdbc.internal.JdbcCoordinatorImpl.coordinateWork(JdbcCoordinatorImpl.java:306) [hibernate-core-5.jar:5.3.25.Final]
	at org.hibernate.internal.SessionImpl.doWork(SessionImpl.java:2428) [hibernate-core-5.jar:5.3.25.Final]
	at org.hibernate.internal.SessionImpl.doReturningWork(SessionImpl.java:2424) [hibernate-core-5.jar:5.3.25.Final]
	at com.redhat.rhn.common.db.datasource.CachedStatement.doWithStolenConnection(CachedStatement.java:909) [rhn.jar:?]
	at com.redhat.rhn.common.db.datasource.CachedStatement.executeChecking(CachedStatement.java:454) [rhn.jar:?]
	at com.redhat.rhn.common.db.datasource.CachedStatement.executeElaboratorBatch(CachedStatement.java:393) [rhn.jar:?]
	at com.redhat.rhn.common.db.datasource.CachedStatement.executeElaboratorBatch(CachedStatement.java:420) [rhn.jar:?]
	at com.redhat.rhn.common.db.datasource.CachedStatement.executeElaborator(CachedStatement.java:369) [rhn.jar:?]
	at com.redhat.rhn.common.db.datasource.SelectMode.elaborate(SelectMode.java:137) [rhn.jar:?]
	at com.redhat.rhn.common.db.datasource.DataResult.elaborate(DataResult.java:148) [rhn.jar:?]
	at com.redhat.rhn.manager.BaseManager.processPageControl(BaseManager.java:201) [rhn.jar:?]
	at com.redhat.rhn.manager.BaseManager.makeDataResult(BaseManager.java:58) [rhn.jar:?]
	at com.redhat.rhn.manager.BaseManager.makeDataResult(BaseManager.java:76) [rhn.jar:?]
	at com.redhat.rhn.manager.system.SystemManager.registeredList(SystemManager.java:1153) [rhn.jar:?]
	at com.redhat.rhn.frontend.action.renderers.RecentSystemsRenderer.render(RecentSystemsRenderer.java:45) [rhn.jar:?]
	at com.redhat.rhn.frontend.action.renderers.BaseFragmentRenderer.renderAsync(BaseFragmentRenderer.java:50) [rhn.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.directwebremoting.impl.CreatorModule$1.doFilter(CreatorModule.java:178) [dwr.jar:?]
	at org.directwebremoting.impl.CreatorModule.executeMethod(CreatorModule.java:184) [dwr.jar:?]
	at org.directwebremoting.impl.DefaultRemoter.execute(DefaultRemoter.java:353) [dwr.jar:?]
	at org.directwebremoting.impl.DefaultRemoter.execute(DefaultRemoter.java:306) [dwr.jar:?]
	at org.directwebremoting.dwrp.BaseCallHandler.handle(BaseCallHandler.java:110) [dwr.jar:?]
	at org.directwebremoting.servlet.UrlProcessor.handle(UrlProcessor.java:211) [dwr.jar:?]
	at org.directwebremoting.servlet.UrlProcessor.handle(UrlProcessor.java:185) [dwr.jar:?]
	at org.directwebremoting.servlet.DwrServlet.doPost(DwrServlet.java:144) [dwr.jar:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:660) [tomcat-servlet-4.0-api.jar:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:741) [tomcat-servlet-4.0-api.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) [catalina.jar:9.0.36]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:9.0.36]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53) [tomcat-websocket.jar:9.0.36]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:9.0.36]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:9.0.36]
	at com.opensymphony.module.sitemesh.filter.PageFilter.parsePage(PageFilter.java:142) [sitemesh.jar:?]
	at com.opensymphony.module.sitemesh.filter.PageFilter.doFilter(PageFilter.java:58) [sitemesh.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:9.0.36]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:9.0.36]
	at com.redhat.rhn.frontend.servlets.LocalizedEnvironmentFilter.doFilter(LocalizedEnvironmentFilter.java:69) [rhn.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:9.0.36]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:9.0.36]
	at com.redhat.rhn.frontend.servlets.EnvironmentFilter.doFilter(EnvironmentFilter.java:100) [rhn.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:9.0.36]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:9.0.36]
	at com.redhat.rhn.frontend.servlets.SessionFilter.doFilter(SessionFilter.java:56) [rhn.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:9.0.36]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:9.0.36]
	at com.redhat.rhn.frontend.servlets.SetCharacterEncodingFilter.doFilter(SetCharacterEncodingFilter.java:96) [rhn.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:9.0.36]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:9.0.36]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202) [catalina.jar:9.0.36]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96) [catalina.jar:9.0.36]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:541) [catalina.jar:9.0.36]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139) [catalina.jar:9.0.36]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92) [catalina.jar:9.0.36]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:690) [catalina.jar:9.0.36]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74) [catalina.jar:9.0.36]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343) [catalina.jar:9.0.36]
	at org.apache.coyote.ajp.AjpProcessor.service(AjpProcessor.java:432) [tomcat-coyote.jar:9.0.36]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) [tomcat-coyote.jar:9.0.36]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:868) [tomcat-coyote.jar:9.0.36]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1591) [tomcat-coyote.jar:9.0.36]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) [tomcat-coyote.jar:9.0.36]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:9.0.36]
	at java.lang.Thread.run(Thread.java:829) [?:?]
----
```

### Additional information

I had to revert the snapshot and currently we are back at 2022.08, but I have collected a few logs. Before I put a lot of (possibly useless) information here, let me know what to provide.

**Comment by mlueckert** [comment]
Looks like I have a similar issue. We are using SUSE Manager and I did the upgrade from 4.2.10 to 4.3.4.
The upgrade was successful but I am now unable to login to the WebUI.

These are the interesting log entries:

rhn_web_ui.log

2023-02-21 11:23:44,356 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-3] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 0, SQLState: 42P01
2023-02-21 11:23:44,357 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-3] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - ERROR: relation "hibernate_sequence" does not exist
  Position: 17
2023-02-21 11:23:44,357 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-3] ERROR com.redhat.rhn.frontend.servlets.SessionFilter - Error during transaction. Rolling back
org.hibernate.exception.SQLGrammarException: could not extract ResultSet


postgres logfile
2023-02-21 12:12:00.082 CET susemanager susemanager [13423]STATEMENT:  select 'c3p0 ping' from dual
2023-02-21 12:12:00.106 CET susemanager susemanager [13151]ERROR:  relation "hibernate_sequence" does not exist at character 17
2023-02-21 12:12:00.106 CET susemanager susemanager [13151]STATEMENT:  select nextval ('hibernate_sequence')
2023-02-21 12:12:00.107 CET susemanager susemanager [13151]ERROR:  current transaction is aborted, commands ignored until end of transaction block
2023-02-21 12:12:00.107 CET susemanager susemanager [13151]STATEMENT:  select 'c3p0 ping' from dual

I have opened a SUSE case now as we have official support.

Best regards, Marc

**Comment by mlueckert** [comment]
I was able to solve my issue with the help of SUSE support. After the successful "zypper migration", I had to perform a "zypper patch". This updated some additional packages and finally solved it.

This is the original mail from SUSE support:

> Thank you for the detailed problem description, it would have been good to be able to look at a supportconfig from the SUMA server generated while in its problem state. However, we actually had a similar case in the past with the same error and problem not being able to login after upgrading to 4.3.4.

> This was due to some spacewalk-java packages not having been properly updated during the upgrade and having a version mismatch, eg.

 > spacewalk-java
 > spacewalk-java-config
 > spacewalk-java-lib

> The solution in that case should be to simply after the upgrade to verify all the spacewalk-java versions match, and if not also run "zypper patch" (or "zypper up") to make sure all packages are updated, then restart postgresql and the spacewalk services.

> For completeness sake the current latest "spacewalk-java" package versions at the time of writing this email for SUMA 4.3.4 should be:

 > spacewalk-java-lib-4.3.46-150400.3.28.1.noarch
 > spacewalk-java-postgresql-4.3.46-150400.3.28.1.noarch
 > spacewalk-java-4.3.46-150400.3.28.1.noarch
 > spacewalk-java-config-4.3.46-150400.3.28.1.noarch

Hope this helps others who have a similar issue.

**Comment by umrubi1** [comment]
I thanks for provide this info. In my case, I am already on OSL 15.4 . 
In your case, you have upgraded from SLES 15.3 to 15.4 (4.2x to 4.3.x), but I`ll give it a try next week.

**Comment by mcalmer** [comment]
@umrubi1 did you check your installation? Is it fixed now?

**Comment by parlt91** [comment]
@umrubi1 Looks like the initial issue was resolved. I am closing this one.
If your `autoinstallation` issues still persist please open a new one for it.

**Comment by umrubi1** [comment]
Sorry .. to much to do in parallel. I will create an new ticket later. For the documentation:

## Initial state
* Do not forget to create an snapshot (if possible)!

```
# zypper info Uyuni-server-release

Loading repository data...
Reading installed packages...

Information for package Uyuni-Server-release:
---------------------------------------------
Repository     : Uyuni Server Stable
Name           : Uyuni-Server-release
Version        : 2023.03-220400.199.1.uyuni2
Arch           : x86_64
Vendor         : obs://build.opensuse.org/systemsmanagement:Uyuni
Support Level  : Level 3
Installed Size : 1.4 KiB
Installed      : Yes
Status         : out-of-date (version 2022.08-185.2.uyuni2 installed)
Source package : Uyuni-Server-release-2023.03-220400.199.1.uyuni2.src
Summary        : Uyuni Server
Description    :
    Uyuni lets you efficiently manage physical, virtual,
    and cloud-based Linux systems. It provides automated and cost-effective
    configuration and software management, asset management, and system
    provisioning.
```
## Update
* Stop services
```
# spacewalk-service stop
```
* Update zypper binary
```
# zypper ref
# zypper in zypper
# rpm -qa|grep zypper

### zypper version is 1.14.57 or newer
# zypper-1.14.59-150400.3.12.2.x86_64
```
* Update the packages
```
# zypper up --allow-vendor-change

....
The following pattern is going to be upgraded:
  uyuni_server

The following product is going to be upgraded:
  Uyuni Server  2022.08-0 -> 2023.03-0

The following package is going to change architecture:
  netty  noarch -> x86_64

The following 4 packages are going to change vendor:
  golang-github-QubitProducts-exporter_exporter  SUSE LLC <https://www.suse.com/> -> obs://build.opensuse.org/systemsmanagement:Uyuni
  guava                                          obs://build.opensuse.org/systemsmanagement:Uyuni -> SUSE LLC <https://www.suse.com/>
  netty                                          obs://build.opensuse.org/systemsmanagement:Uyuni -> SUSE LLC <https://www.suse.com/>
  prometheus-postgres_exporter                   SUSE LLC <https://www.suse.com/> -> obs://build.opensuse.org/systemsmanagement:Uyuni

The following product is going to change vendor:
  Uyuni Server  openSUSE -> obs://build.opensuse.org/systemsmanagement:Uyuni
```

* After download and package installation - check database output of the first start after the update:
```
Mar 30 11:59:19 uyuni1 systemd[1]: Starting Uyuni check database...
¦ The job identifier is 34220.
Mar 30 11:59:20 uyuni1 spacewalk-startup-helper[68743]: report_db_host = uyuni1.vi.vector.int
Mar 30 11:59:59 uyuni1 spacewalk-startup-helper[68755]: Schema upgrade: [susemanager-schema-4.3.13-1.2.uyuni2] -> [susemanager-schema-4.4.5-220400.1.3.uyuni2]
Mar 30 11:59:59 uyuni1 spacewalk-startup-helper[68755]: Searching for upgrade path to: [susemanager-schema-4.4.5-220400]
Mar 30 11:59:59 uyuni1 spacewalk-startup-helper[68755]: Searching for upgrade path to: [susemanager-schema-4.4.5]
Mar 30 11:59:59 uyuni1 spacewalk-startup-helper[68755]: Searching for start path:  [susemanager-schema-4.3.13-1]
Mar 30 11:59:59 uyuni1 spacewalk-startup-helper[68755]: Searching for start path:  [susemanager-schema-4.3.13]
Mar 30 11:59:59 uyuni1 spacewalk-startup-helper[68755]: The path: [susemanager-schema-4.3.13] -> [susemanager-schema-4.4.0] -> [susemanager-schema-4.4.1] -> [susemanager-schema-4.4.2] -> [susemanager-schema-4.4.3] -> [susemanager-schema-4.4.4] -> [susemanager-schema-4.4.5]
Mar 30 11:59:59 uyuni1 spacewalk-startup-helper[68755]: Planning to run schema upgrade with dir '/var/log/spacewalk/schema-upgrade/schema-from-20230330-115959'
Mar 30 11:59:59 uyuni1 spacewalk-startup-helper[68755]: Executing spacewalk-sql, the log is in [/var/log/spacewalk/schema-upgrade/schema-from-20230330-115959-to-susemanager-schema-4.4.5.log].
Mar 30 12:00:04 uyuni1 spacewalk-startup-helper[68755]: [3.9K blob data]
Mar 30 12:00:04 uyuni1 spacewalk-startup-helper[68755]: The database schema was upgraded to version [susemanager-schema-4.4.5-220400.1.3.uyuni2].
Mar 30 12:00:04 uyuni1 spacewalk-startup-helper[69150]: Schema upgrade: [uyuni-reportdb-schema-4.3.5-1.21.uyuni2] -> [uyuni-reportdb-schema-4.4.2-220400.1.3.uyuni2]
Mar 30 12:00:04 uyuni1 spacewalk-startup-helper[69150]: Searching for upgrade path to: [uyuni-reportdb-schema-4.4.2-220400]
Mar 30 12:00:04 uyuni1 spacewalk-startup-helper[69150]: Searching for upgrade path to: [uyuni-reportdb-schema-4.4.2]
Mar 30 12:00:04 uyuni1 spacewalk-startup-helper[69150]: Searching for start path:  [uyuni-reportdb-schema-4.3.5-1]
Mar 30 12:00:04 uyuni1 spacewalk-startup-helper[69150]: Searching for start path:  [uyuni-reportdb-schema-4.3.5]
Mar 30 12:00:04 uyuni1 spacewalk-startup-helper[69150]: The path: [uyuni-reportdb-schema-4.3.5] -> [uyuni-reportdb-schema-4.4.0] -> [uyuni-reportdb-schema-4.4.1] -> [uyuni-reportdb-schema-4.4.2]
Mar 30 12:00:05 uyuni1 spacewalk-startup-helper[69150]: Planning to run schema upgrade with dir '/var/log/spacewalk/reportdb-schema-upgrade/schema-from-20230330-120004'
Mar 30 12:00:05 uyuni1 spacewalk-startup-helper[69150]: Executing spacewalk-sql, the log is in [/var/log/spacewalk/reportdb-schema-upgrade/schema-from-20230330-120004-to-uyuni-reportdb-schema-4.4.2.log].
Mar 30 12:00:05 uyuni1 spacewalk-startup-helper[69150]: [157B blob data]
Mar 30 12:00:05 uyuni1 spacewalk-startup-helper[69150]: The database schema was upgraded to version [uyuni-reportdb-schema-4.4.2-220400.1.3.uyuni2].
Mar 30 12:00:05 uyuni1 systemd[1]: Finished Uyuni check database.
¦¦ Subject: A start job for unit uyuni-check-database.service has finished successfully
¦¦ Defined-By: systemd
¦¦ Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
¦¦
¦¦ A start job for unit uyuni-check-database.service has finished successfully.
¦¦
¦¦ The job identifier is 34220.
```

* Login to Uyuni UI 
* Restart the system
* After everything was coming up, check the avaliable updates of you Uyuni Proxies (if you have some like me) and update them too. As long, they are on the same (old) software level, some "proxy related errors" will be shown in the logs, followed by an ID. In my case it where the Uyuni System IDs of my proxy systems.

## Issues after that
* I have some autoinstallation profiles. After starting an new virtual machine, only one profile is shown
* Have found out, that the rest will be shown, during cobbler is doing some tasks (e.g. by clicking on "Update" in UI and/or sync jobs between the proxies). 
*  Will open an new ticket for that, should not be the focus here.
